primitives/manta/src/xcm.rs

```diff
@@ -446,3 +452,170 @@ impl<
         Ok((asset_id, amount, receiver))
     }
 }
+
+// 4_000_000_000 is a typical configuration value provided to dApp developers for `dest_weight`
+// argument when sending xcm message to Calamari. ie moonbeam, sub-wallet, phala, etc
+pub const ADVERTISED_DEST_WEIGHT: u64 = 4_000_000_000;
+
+// Composition of self_reserve message composed by xTokens on the sender side
+pub fn self_reserve_xcm_message_receiver_side<T>() -> Xcm<T> {
```

Comment:
@ghzlatarev @ferrell-code pointed out that this code should be behind a testing feature to not blow up runtime wasm size

Reply:
postponed to #855